src/actix/api/delete_api.rs

```diff
@@ -0,0 +1,50 @@
+use crate::actix::helpers::process_response;
+use actix_web::rt::time::Instant;
+use actix_web::web::Query;
+use actix_web::{delete, web, Responder};
+use collection::operations::point_ops::PointOperations;
+use collection::operations::types::{DeleteRequest, UpdateResult};
+use collection::operations::CollectionUpdateOperations;
+use schemars::JsonSchema;
+use serde::{Deserialize, Serialize};
+use std::sync::Arc;
+use storage::content_manager::errors::StorageError;
+use storage::content_manager::toc::TableOfContent;
+
+#[derive(Deserialize, Serialize, JsonSchema)]
+pub struct DeleteParam {
+    pub wait: Option<bool>,
+}
+
+async fn do_delete_points(
+    toc: &TableOfContent,
+    collection_name: &str,
+    request: DeleteRequest,
+    wait: bool,
+) -> Result<UpdateResult, StorageError> {
+    toc.update(
+        collection_name,
+        CollectionUpdateOperations::PointOperation(PointOperations::DeletePointsByFilter(
+            request.filter,
+        )),
+        wait,
+    )
+    .await
+}
+
+#[delete("/collections/{name}/points")]
```

Comment:
In future we also plan to have individual REST endpoints for all point operation including regular point removing by ids, so it might be better to use more specific path name for this operation. E.g. 
```suggestion
#[delete("/collections/{name}/points/delete_by_filter")]
```

Reply:
how about using: `#[delete("/collections/{name}/points/by_filter")]` or `#[delete("/collections/{name}/points/filtered")]`. The delete word in the url's path could be different than the actual HTTP verb. I'd do some agnostic stuff here, always but thats up to you

Reply:
`#[delete("/collections/{name}/points/by_filter")]` would be good, agreed

Reply:
Alright! Will do so!